lnwallet/channel.go

```diff
@@ -6416,6 +6570,16 @@ func (lc *LightningChannel) ReceiveHTLC(htlc *lnwire.UpdateAddHTLC) (uint64, err
 		BlindingPoint: htlc.BlindingPoint,
 	}
 
+	if len(htlc.CustomRecords) > 0 {
```

Comment:
If `ExtraData` contains a protocol-range TLV that _isn't_ blinding point, all of our `htlc.CustomRecords` will be overwritten because we'll hit `ExtraData != nil`? 

While we're not expecting this (there aren't any other defined protocol TLVs for `update_add_htlc` right now), I think we should account for the possibility that other nodes in the network may relay these fields in the future because as-is it'll break our code.

Reply:
I think this has been addressed by not touching the `ExtraData` field unless we're serializing for wire or storage.